packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatContext } from './ChatContext';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatError } from './ChatClientState';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatMessageWithStatus } from './types/ChatMessageWithStatus';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { CommunicationTokenCredential } from '@azure/communication-common';
+
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+/**
+ * @private
+ */
+export class ResourceDownloadQueue {
+  private _messageQueue: ChatMessageWithStatus[] = [];
```

Comment:
Maybe name `_messagesNeedingResourceRetrieval` or similar?

Reply:
Changed the name